src/rpc/server.cpp

```diff
@@ -79,16 +79,20 @@ void RPCTypeCheck(const UniValue& params,
             break;
 
         const UniValue& v = params[i];
-        if (!((v.type() == t) || (fAllowNull && (v.isNull()))))
-        {
-            string err = strprintf("Expected type %s, got %s",
-                                   uvTypeName(t), uvTypeName(v.type()));
-            throw JSONRPCError(RPC_TYPE_ERROR, err);
+        if (!(fAllowNull && v.isNull())) {
+            RPCTypeCheckArgument(v, t);
         }
         i++;
     }
 }
 
+void RPCTypeCheckArgument(const UniValue& value, UniValue::VType typeExpected)
```

Comment:
This seems redundant. Type-checking is the purpose of the .get_&lt;type> methods on UniValue.